x-pack/legacy/plugins/epm/common/types.ts

```diff
@@ -108,28 +124,19 @@ export interface Dataset {
   type: string;
 }
 
-export interface RegistryPackage {
-  name: string;
-  version: string;
-  description: string;
-  readme?: string;
-  icon: string;
-  requirement: RequirementsByServiceName;
-  title?: string;
-  screenshots?: ScreenshotItem[];
-  assets: string[];
-  datasets: Dataset[];
+// some properties are optional in Registry responses but required in EPM
+// internal until we need them
+interface PackageAdditions {
+  title: string;
+  assets: AssetsGroupedByServiceByType;
 }
 
 // Managers public HTTP response types
 export type PackageList = PackageListItem[];
 
-export type PackageListItem = Installable<Required<RegistryListItem>>;
+export type PackageListItem = Installable<RegistryListItem & PackageAdditions>;
 export type PackagesGroupedByStatus = Record<InstallationStatus, PackageList>;
-
-export type PackageInfo = Installable<
-  Required<RegistryPackage> & { assets: AssetsGroupedByServiceByType }
->;
+export type PackageInfo = Installable<RegistryPackage & PackageAdditions>;
```

Comment:
Remove the `Required<RegistryPackage>` which said all items in `RegistryPackage` were required, which is not true. Any items which are optional in `RegistryPackage` are optional in `PackageInfo` (except the changes from `PackageAdditions`)